### ISSUE TITLE
mavenLocal broken when <localRepository>/my-relocated-maven-repo</localRepository> found in Maven settings.xml

### ISSUE THREAD

**Issue by avelella** [gradle/gradle#2843]
Regression in 4.1 compared to 3.4.1 - mavenLocal() does not work

### Expected Behavior
Dependencies or plugins can't be read from mavenLocal(), where maven local is configured in my settings.xml to be in a custom location.

### Current Behavior
I can publish to mavenLocal() and it goes to my custom repository. I can't read from mavenLocal(). Gradle does see my custom settings.xml, but it's using the current directory as the base. E.g.: (current-directory)/my-relocated-maven-repo

This looks a lot like the issue reported here: https://github.com/gradle/gradle/issues/1811

But that issue was using the empty tag: `<localRepository/>`

### Context
We can't use mavenLocal in our builds.

### Steps to Reproduce (for bugs)
Create a file ~/.m2/settings.xml

Contents:
```
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
    <localRepository>/my-relocated-maven-repo</localRepository> 
</settings>

```
### Your Environment
Windows. Gradle 4.1.

**Comment by bigdaz** [comment]
In Gradle 3.5, we improved performance by removing the need to spawn an external process to read the Maven `settings.xml` file: https://github.com/gradle/gradle/commit/547a80232622b9886d518e2807db8b1e6c0434f6. In doing so, we switched from using `DefaultSettingsBuilder` to use the lighter-weight `DefaultSettingsReader` (both Maven types).

In doing so, it looks like we lost [this special handling](https://github.com/apache/maven/blob/master/maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java#L106-L115) of absolute file paths on Windows: 
```
// for the special case of a drive-relative Windows path, make sure it's absolute to save plugins from trouble
String localRepository = userSettings.getLocalRepository();
if ( localRepository != null && localRepository.length() > 0 )
{
    File file = new File( localRepository );
    if ( !file.isAbsolute() && file.getPath().startsWith( File.separator ) )
    {
        userSettings.setLocalRepository( file.getAbsolutePath() );
    }
}
```
So this is the case where `/file/path` is _not_ an absolute path on Windows, since it's missing the drive prefix.
If we do add this behaviour, we should probably do it _after_ we've done any placeholder replacement here: https://github.com/gradle/gradle/blob/e6598d4bc68c1b3fde2b6136cd78360f27d8e1be/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/mvnsettings/DefaultLocalMavenRepositoryLocator.java#L51-L51

@avelella would you be interested in contributing a fix? We'd need some test coverage in `DefaultLocalMavenRepositoryLocatorTest.groovy`.

In the meantime, I think a workaround would be to provide a fully-qualified file path for windows (with the drive letter).

**Comment by bmuschko** [comment]
@blindpirate Can you please add the milestone so we know later what version of Gradle introduced the fix?

**Comment by blindpirate** [comment]
@bmuschko No problem!

**Comment by blindpirate** [comment]
@bmuschko  Looks like fixed issues are rendered via `https://services.gradle.org/fixed-issues/4.2`, and not written in `notes.md`. I'm wondering where should I write this issue down?

**Comment by bmuschko** [comment]
@blindpirate Never mind about the release notes. For some reason I thought this was a PR.
